tests/foreman/ui_airgun/test_role.py

```diff
@@ -33,6 +36,132 @@ def module_loc():
+    new_role_description = gen_string('alpha')
+    new_org = entities.Organization().create()
+    new_loc = entities.Location(organization=[new_org]).create()
+    with session:
+        session.role.create({
+            'name': role_name,
+            'description': role_description,
+            'organizations.assigned': [module_org.name],
+            'locations.assigned': [module_loc.name]
+        })
+        values = session.role.read(role_name)
+        assert values['name'] == role_name
+        assert values['description'] == role_description
+        assert values['organizations']['assigned'] == [module_org.name]
+        assert values['locations']['assigned'] == [module_loc.name]
+        session.filter.create(role_name, {
+            'resource_type': resource_type,
+            'permission.assigned': permissions
+        })
+        assigned_permissions = session.filter.read_permissions(role_name)
+        assert set(assigned_permissions[resource_type]) == set(permissions)
+        # clone the role
+        session.role.clone(role_name, {'name': cloned_role_name})
+        assert session.role.search(cloned_role_name)[0]['Name'] == cloned_role_name
+        assigned_permissions = session.filter.read_permissions(cloned_role_name)
+        assert list(assigned_permissions.keys()) == [resource_type]
+        assert set(assigned_permissions[resource_type]) == set(permissions)
+        # update the role
+        session.role.update(role_name, {
+            'name': new_role_name,
+            'description': new_role_description,
+            'organizations.assigned': [new_org.name],
+            'locations.assigned': [new_loc.name]
+        })
+        assert not session.role.search(role_name)
+        values = session.role.read(new_role_name)
+        assert values['name'] == new_role_name
+        assert values['description'] == new_role_description
+        assert set(values['organizations']['assigned']) == {module_org.name, new_org.name}
+        assert set(values['locations']['assigned']) == {module_loc.name, new_loc.name}
+        # delete the role
+        session.role.delete(new_role_name)
+        assert not session.role.search(new_role_name)
+        # delete the cloned role
+        session.role.delete(cloned_role_name)
+        assert not session.role.search(cloned_role_name)
+
+
+@tier2
+def test_positive_assign_cloned_role(session):
+    """Clone role and assign it to user
+
+    :id: cbb17f37-9039-4875-981b-1f427b095ed1
+
+    :customerscenario: true
+
+    :expectedresults: User is created successfully
+
+    :BZ: 1353788
+
+    :CaseImportance: Critical
+    """
+    role_name = random.choice(ROLES)
+    cloned_role_name = gen_string('alpha')
+    user_name = gen_string('alpha')
+    user_password = gen_string('alpha')
+    with session:
+        session.role.clone(role_name, {'name': cloned_role_name})
+        assert session.role.search(cloned_role_name)[0]['Name'] == cloned_role_name
```

Comment:
We can make this scenario better - not just ensure some role with proper name was created (such scenario isn't that different from `create` test), but read values in original role, then clone, read cloned values, ensure they're the same. Such flow will verify cloning actually working.

Reply:
The clone work only for filters (create a new role with cloning the filters), new values should be entered for the clone.

Reply:
Well, you got my point - if only filters are cloned, then check filters instead. Idea is to ensure cloning works, not just some new empty entity was created.

Reply:
@abalakh This is a bug coverage test (that has a concrete scenario), there is other test where we verify that the cloned role has that permissions.